src/modules/flow/http-server/http-server.c

```diff
@@ -214,79 +193,100 @@ boolean_close(struct sol_flow_node *node, void *data)
+        sol_flow_node_get_type(node);
 
-    SOL_HTTP_PARAM_FOREACH_IDX (sol_http_request_get_params(request), value, idx) {
-        switch (value->type) {
-        case SOL_HTTP_PARAM_POST_FIELD:
-            if (streq(value->value.key_value.key, "value")) {
-                free(mdata->value.s);
-                mdata->value.s = strdup(value->value.key_value.value);
-                if (!mdata->value.s) {
-                    r = -ENOMEM;
-                    SOL_WRN("Could not duplicate the string");
-                    goto end;
-                }
-
-                r = sol_http_server_set_last_modified(server, mdata->path, time(NULL));
-                SOL_INT_CHECK_GOTO(r, < 0, end);
-            }
-            break;
-        case SOL_HTTP_PARAM_HEADER:
-            if (streq(value->value.key_value.key, HTTP_HEADER_ACCEPT)) {
-                if (strstr(value->value.key_value.value, HTTP_HEADER_CONTENT_TYPE_JSON))
-                    send_json = true;
-            }
-            break;
-        default:
-            break;
-        }
+    r = type->process_cb(mdata, packet);
+    SOL_INT_CHECK(r, < 0, r);
+
+    r = sol_http_server_set_last_modified(server, mdata->path, time(NULL));
+    SOL_INT_CHECK(r, < 0, r);
+
+    return 0;
+}
+
+static int
+boolean_post_cb(struct http_data *mdata, struct sol_flow_node *node,
+    struct sol_http_param_value *value)
+{
+    if (streq(value->value.key_value.value, "true"))
+        mdata->value.b = true;
+    else if (streq(value->value.key_value.value, "false"))
+        mdata->value.b = false;
+    else
+        return -EINVAL;
+
+    sol_flow_send_boolean_packet(node,
+            SOL_FLOW_NODE_TYPE_HTTP_SERVER_BOOLEAN__OUT__OUT,
+            mdata->value.b);
+
+    return 0;
+}
+
+static int
+boolean_response_cb(struct http_data *mdata, struct sol_buffer *content, bool json)
+{
+    int r;
+    char response_str[128];
+
+    if (json) {
+        r = snprintf(response_str, sizeof(response_str), "{\"%s\": %s}", mdata->path,
+            mdata->value.b == true ? "true" : "false");
+    } else {
+        r = snprintf(response_str, sizeof(response_str), "%s", mdata->value.b == true ? "true" : "false");
     }
+    SOL_INT_CHECK(r, < 0, r);
+
+    r = sol_buffer_set_slice(content, sol_str_slice_from_str(response_str));
+    SOL_INT_CHECK(r, < 0, r);
+
+    return 0;
+}
 
-    if (send_json) {
+static int
+boolean_process_cb(struct http_data *mdata, const struct sol_flow_packet *packet)
+{
+    return sol_flow_packet_get_boolean(packet, &mdata->value.b);
+}
+
+/* ------------------------------------- string ------------------------------------------- */
+
+static int
+string_response_cb(struct http_data *mdata, struct sol_buffer *content, bool json)
+{
+    int r = 0;
+    char *response_str;
+
+    if (json) {
         char *escaped_str;
         size_t escaped_len = sol_json_calculate_escaped_string_len(mdata->value.s);
         escaped_str = malloc(escaped_len);
```

Comment:
use the buffer potential! :-)

If you use the `type->json_append_value()` as I said before, this function would be:

```c
if (json) {
   sol_buffer_reset(buffer);
   r = sol_buffer_append_printf(buffer, "{\"%s\":", mdata->path); // see this is similar to the boolean? This could be in the caller function! :-)
   r = type->json_append_value(mdata, content);
   SOL_INT_CHECK(r, < 0, r);
   r = sol_buffer_append_slice(content, sol_str_slice_from_str("}"));
   SOL_INT_CHECK(r, < 0, r);
}
```

With that `json_append_value` being implemented as:
```c
escaped_len = sol_json_calculate_escaped_string_len(mdata->value.s)
r = sol_buffer_ensure(buffer, buffer->used + 2 + escaped_len);
SOL_INT_CHECK(r, < 0, r);
r = sol_buffer_append_slice(buffer, sol_str_slice_from_string("\"")); // we should have a sol_buffer_append_char() and sol_buffer_insert_char()!
SOL_INT_CHECK(r, < 0, r);
sol_json_escape_string(mdata->value.s, sol_buffer_at_end(buffer), escaped_len);
r = sol_buffer_append_slice(buffer, sol_str_slice_from_string("\"")); // we should have a sol_buffer_append_char() and sol_buffer_insert_char()!
SOL_INT_CHECK(r, < 0, r);
```